src/sidebar/components/MarkdownEditor.tsx

```diff
@@ -176,24 +184,69 @@ function ToolbarButton({
 type TextAreaProps = {
   classes?: string;
   containerRef?: Ref<HTMLTextAreaElement>;
+  atMentionsEnabled: boolean;
 };
 
 function TextArea({
   classes,
   containerRef,
+  atMentionsEnabled,
   ...restProps
 }: TextAreaProps & JSX.TextareaHTMLAttributes<HTMLTextAreaElement>) {
+  const [popoverOpen, setPopoverOpen] = useState(false);
+  const textareaRef = useSyncedRef(containerRef);
+
+  useEffect(() => {
+    if (!atMentionsEnabled) {
+      return () => {};
+    }
+
+    const textarea = textareaRef.current!;
+    const listenerCollection = new ListenerCollection();
+
+    // We listen for `keyup` to make sure the text in the textarea reflects the
+    // just-pressed key when we evaluate it
+    listenerCollection.add(textarea, 'keyup', e => {
```

Comment:
The suggestions popup does not currently appear if the caret is repositioned by clicking in the middle of a mention. I haven't checked, but I'd guess that also applies to using the caret movement UIs on mobile. I think that's fine for this PR but you could instead use the `selectionchange` event. The downside is that I don't think that event provides a way to determine why the selection changed.

Reply:
Good point. There's another thing to take into consideration actually, which is that the popover is closed when clicking away, so I'll probably try to address that separately to try different combinations and even handlers.